python/llm/src/ipex_llm/transformers/models/phi3.py

```diff
@@ -137,38 +137,39 @@ def attention_forward(
         key_states, value_states = past_key_value.update(key_states, value_states,
                                                          self.layer_idx, None)
 
-    if (isinstance(past_key_value, DynamicFp8Cache) and
-            use_sdp_fp8(q_len, kv_seq_len, query_states)):
+    if use_sdp(q_len, kv_seq_len, self.head_dim, query_states):
         import linear_q4_0
-        attn_output = linear_q4_0.sdp_fp8(query_states, key_states, value_states, attention_mask)
-    elif (isinstance(past_key_value, DynamicNormalCache) and
-            use_sdp(q_len, kv_seq_len, self.head_dim, query_states)):
+        if isinstance(past_key_value, DynamicFp8Cache):
+            attn_output = linear_q4_0.sdp_fp8(query_states, key_states, value_states,
+                                              attention_mask)
+        else:
+            attn_output = linear_q4_0.sdp(query_states, key_states, value_states, attention_mask)
+    elif use_sdp_causal(q_len, kv_seq_len, query_states, self.training):
```

Comment:
maybe need to judge head_dim here?

Reply:
good, will check `head_dim` in `use_sdp_causal` in next PR